votes/0625-new-triage-method-for-cherrypick.md

```diff
@@ -0,0 +1,54 @@
+
+When a version is triaged, the triager needs to remove the corresponding `may-affect-x.x` label. If the version is affected, the triager needs to add a corresponding `affects-x.x` label to the issue and in the meanwhile the `may-affect-x.x` label can be automatically removed by the ti-chi-bot, otherwise the triager can simply remove the `may-affect-x.x` label. So when a issue has a label `may-affect-x.x`, this means the issue has not been diagnosed on version x.x. When a issue has a label `affects-x.x`, this means the issue has been diagnosed on version x.x and identified affected. When both the two labels are missing, this means the issue has been diagnosed on version x.x but the version is not affected.
+
+The status of the affection of a certain issue can be then determined by the combination of the existence of the corresponding `may-affect-x.x` and `affect-x.x` labels on the issue, see the table bellow for a clearer illustration.
+
+| may-affect-x.x | affects-x.x |                         status                                |
+|:--------------:|:-----------:|:-------------------------------------------------------------:|
+|     YES        |    NO       | version x.x has not been diagnosed                            |
+|     NO         |    NO       | version x.x has been diagnosed and identified as not affected |
+|     NO         |    YES      | version x.x has been diagnosed and identified as affected     |
+|     YES        |    YES      | invalid status                                                |
+
+In such a workflow, we can assure all the effected versions could be correctly diagnosed. 
+
+For pull requests, we will introduce a new checker named `check-issue-triage-complete`. The checker will first check whether the [corresponding issue](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/contribute-code.html#referring-to-an-issue) has any `type/xx` label, if not, the checker fails. Then for issues with `type/bug` label, there must also exist a `severity/xx` label, otherwise, the checker fails. For `type/bug` issue with `severity/critical` or `severity/major` label, the checker checks if there is any `may-affect-x.x` label, which means the issue has not been diagnosed on all needed versions. If there is, the pull request is blocked and not able to be merged. So in order to merge a bugfix pull request into target branch, every other effective version needs to first be diagnosed. Contributors could comment `/run-check-issue-triage-complete` like other checkers to run the checker and update the status.
```

Comment:
Some details to add.

```suggestion
For pull requests, we will introduce a new checker named `check-issue-triage-complete`. The checker will first check whether the [corresponding issue](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/contribute-code.html#referring-to-an-issue) has any `type/xx` label, if not, the checker fails. Then for issues with `type/bug` label, there must also exist a `severity/xx` label, otherwise, the checker fails. For `type/bug` issue with `severity/critical` or `severity/major` label, the checker checks if there is any `may-affect-x.x` label, which means the issue has not been diagnosed on all needed versions. If there is, the pull request is blocked and not able to be merged. So in order to merge a bugfix pull request into the target branch, every other effective version needs to first be diagnosed. 

The bot will automatically trigger the checker to run on the associated PR by listening to the labeled / unlabeled event of `may-affects-x.y` labels on bug issue, contributors also could comment `/run-check-issue-triage-complete` like other checkers to rerun the checker manually and update the status. In addition, if the checker fails, the robot will add the `do-not-merge/needs-triage-completed` label to the PR at the same time, which will be used by other plugins like [tars](https://book.prow.tidb.io/#/en/plugins/tars).
```